lib/matplotlib/dviread.py

```diff
@@ -792,57 +824,52 @@ class PsfontsMap(object):
     file names.
 
     A texmf tree typically includes mapping files called e.g.
-    psfonts.map, pdftex.map, dvipdfm.map. psfonts.map is used by
-    dvips, pdftex.map by pdfTeX, and dvipdfm.map by dvipdfm.
-    psfonts.map might avoid embedding the 35 PostScript fonts (i.e.,
-    have no filename for them, as in the Times-Bold example above),
-    while the pdf-related files perhaps only avoid the "Base 14" pdf
-    fonts. But the user may have configured these files differently.
+    :file:`psfonts.map`, :file:`pdftex.map`, or :file:`dvipdfm.map`.
+    The file :file:`psfonts.map` is used by :program:`dvips`,
+    :file:`pdftex.map` by :program:`pdfTeX`, and :file:`dvipdfm.map`
+    by :program:`dvipdfm`. :file:`psfonts.map` might avoid embedding
+    the 35 PostScript fonts (i.e., have no filename for them, as in
+    the Times-Bold example above), while the pdf-related files perhaps
+    only avoid the "Base 14" pdf fonts. But the user may have
+    configured these files differently.
     """
-    __slots__ = ('_font',)
+    __slots__ = ('_font', '_filename')
 
     def __init__(self, filename):
         self._font = {}
-        with open(filename, 'rt') as file:
+        self._filename = filename
+        if six.PY3 and isinstance(filename, bytes):
+            self._filename = filename.decode('ascii', errors='replace')
```

Comment:
I think the only current users in our code are the PDF backend and the text2path code, both of which just pass in the location of "pdftex.map". I initially thought this might need to be made customizable but I've never seen that as a feature request.